addons/variable-manager/userscript.js

```diff
@@ -0,0 +1,306 @@
+export default async function ({ addon, global, console, msg }) {
+  const vm = addon.tab.traps.vm;
+
+  let contentArea;
+  let localVariables = [];
+  let globalVariables = [];
+  let preventUpdate = false;
+
+  const manager = document.createElement("div");
+  manager.classList.add(addon.tab.scratchClass("asset-panel_wrapper"), "sa-var-manager");
+
+  const searchBox = document.createElement("input");
+  searchBox.placeholder = msg("search");
+  searchBox.className = "sa-var-manager-searchbox";
+
+  searchBox.addEventListener("input", (e) => {
+    for (const variable of localVariables) {
+      variable.handleSearch(searchBox.value);
+    }
+    for (const variable of globalVariables) {
+      variable.handleSearch(searchBox.value);
+    }
+    updateHeadingVisibility();
+  });
+
+  manager.appendChild(searchBox);
+
+  // begin feedback button stuff (remove when out of beta)
+
+  const feedbackWrapper = document.createElement("span");
+  feedbackWrapper.className = "sa-var-manager-feedback";
+
+  const feedbackDesc = document.createElement("span");
+  feedbackDesc.className = "sa-var-manager-feedback-description";
+  feedbackDesc.innerText = msg("feedback-description");
+
+  const feedbackButton = document.createElement("a");
+  feedbackButton.className = "sa-var-manager-feedback-button";
+  feedbackButton.innerText = msg("feedback");
+  feedbackButton.href = "https://scratchaddons.com/feedback?version=variable-manager";
```

Comment:
Is it possible to add the version number on it, just in case?

Optional: Is it possible to use flex so the feedback text is exactly on the right of the search box?

Reply:
i believe w_l said on the discord that addons don't know the SA version number but i guess it wouldn't be too hard to implement it into the addon api.

as for the feedback text, i'm not really too concerned about it because it's probably going to be removed in a few weeks. but i'll experiment/

Reply:
> as for the feedback text, i'm not really too concerned about it because it's probably going to be removed in a few weeks. but i'll experiment/

I mean we got two weeks minimum before the beta tag is removed, so yeah.